src/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/common/SearchResults.cs

```diff
@@ -71,17 +71,18 @@ public class SearchResultReferenceCollection : ReadOnlyCollectionBase
     public class SearchResultEntry
     {
         private DirectoryControl[] _resultControls = null;
+        private SearchResultAttributeCollection _attributes = new SearchResultAttributeCollection();
```

Comment:
Oops. I see this was introduced by 9b4ef7f8a3955798a225577a7baddfca2e10c8c7

Can you add a test? Clearly we don't have one. Even if it would only actually do anything when we run tests locally (since no domain in automated runs)

Reply:
Why not just fix it by changing:
```C#
public SearchResultAttributeCollection Attributes { get; }
```
to be:
```C#
public SearchResultAttributeCollection Attributes { get; } = new SearchResultAttributeCollection();
```
?

Reply:
Or, better yet, by changing the constructors to be:
```C#
internal SearchResultEntry(string dn) : this(dn, new SearchResultAttributeCollection()) { }

internal SearchResultEntry(string dn, SearchResultAttributeCollection)
{
    DistinguishedName = dn;
    Attributes = attrs;
}
```
?  Then we're not allocating a collection that's just going to get overwritten if one is provided to the ctor, and we still get to keep the (implicit) field as readonly.